feature-utils/poly-look/src/react-components/overlays/sideSlider.jsx

```diff
@@ -0,0 +1,422 @@
+    observer.observe(elem);
+  }
+
+  useEffect(() => {
+    if (open) {
+      updateStep(steps.in);
+    } else updateStep(steps.out);
+  }, [open]);
+
+  useEffect(() => {
+    if (step == steps.in) {
+      updateContentsStyle({
+        ...contentsStyle,
+        ...commonStyles.contentsIn,
+        ...commonStyles.contentsTransition,
+        ...commonStyles.duration,
+      });
+      updateBackdropStyle({
+        ...commonStyles.backdropIn,
+        ...commonStyles.backgroundTransition,
+        ...commonStyles.duration,
+        ...commonStyles.noInteraction,
+      });
+    }
+
+    if (step == steps.interactive) {
+      if (!stPoz) {
+        stPoz = contentsRef.current.getBoundingClientRect();
+        alphaInc.x = stPoz.width / ((1 - alpha) * 100);
+        alphaInc.y = stPoz.height / ((1 - alpha) * 100);
+      }
+      updateAnimationsStatus(0);
+      updateBackdropStyle({
+        ...backdropStyle,
+        ...commonStyles.interaction,
+      });
+    }
+
+    if (step == steps.out) {
+      updateContentsStyle({
+        ...contentsStyle,
+        ...commonStyles.duration,
+        ...commonStyles.contentsOutRight,
+      });
+      updateBackdropStyle({
+        ...commonStyles.backdropOut,
+        ...commonStyles.backgroundTransition,
+        ...commonStyles.duration,
+        ...commonStyles.noInteraction,
+      });
+    }
+
+    if (step == steps.outUp) {
+      updateContentsStyle({
+        ...contentsStyle,
+        ...commonStyles.duration,
+        transform: `translate(${leftDistance},${-1 * stPoz.height}px)`,
+      });
+      updateBackdropStyle({
+        ...commonStyles.backdropOut,
+        ...commonStyles.backgroundTransition,
+        ...commonStyles.duration,
+        ...commonStyles.noInteraction,
+      });
+    }
+
+    if (step == steps.outDone) onClose();
+  }, [step]);
+
+  useEffect(() => {
+    if (animationsDoneCount == 2) {
+      if (step == steps.in) updateStep(steps.interactive);
+      else if (step == steps.out || step == steps.outUp)
+        updateStep(steps.outDone);
+    }
+  }, [animationsDoneCount]);
+
+  useEffect(() => {
+    if (axis == axes.x) return;
+
+    if (verticalGesture == "swipe") {
+      updateMovement({ ...movement, swipeY: movement.currentY });
+      if (isPressing) updateChildScroll(commonStyles.noScroll);
+      else updateChildScroll({});
+    } else updateMovement({ ...movement, swipeY: null });
+  }, [verticalGesture]);
+
+  useEffect(() => {
+    if (isPressing)
+      applyStyle(
+        movement.startX - movement.currentX,
+        movement.swipeY
+          ? movement.swipeY - movement.currentY
+          : movement.startY - movement.currentY
+      );
+  }, [movement]);
+
+  function applyStyle(distanceX, distanceY) {
+    let xValue = -1 * distanceX + stPoz.x;
+    let yValue = -1 * distanceY;
+
+    switch (axis) {
+      case axes.x: {
+        updateChildScroll(commonStyles.noScroll);
+        let alphaIncX = Math.floor(distanceX / alphaInc.x) * 0.1;
```

Comment:
This is a local version of the alphaInc global variable, right? But you're splitting it in two different variables here.

Reply:
It's different. The (previously) global was to define something like update segments/intervals based on width and height and how much alpha we get to work with. The local ones are about apply a value based on what segment/interval you are currently in, based on how much you've moved the slider.

Reply:
Then maybe use a different name?

Reply:
I've changed the name, but I'm not 100% confident that I've made a huge difference. So I'm open to better naming suggestions 😅